src/utils/types.ts

```diff
@@ -10,3 +10,18 @@ export type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) ex
 ) => void
   ? I
   : never
+
+export type JSONPrimitive = string | boolean | number | null | undefined
+export type JSONArray = (JSONPrimitive | JSONObject | JSONArray)[]
+export type JSONObject = { [key: string]: JSONPrimitive | JSONArray | JSONObject }
```

Comment:
Top-level primitives and arrays can be valid JSONs. So the correct JSON type definition would be as follows IIUC.

```ts
export type JSONObject = JSONPrimitive | { [key: string]: JSONObject } | JSONObject[]
```

Reply:
You are right. These values can be JSON. I'll make `JSONValue` type like the following and use it.

```ts
type JSONValue = JSONObject | JSONArray | JSONPrimitive
```